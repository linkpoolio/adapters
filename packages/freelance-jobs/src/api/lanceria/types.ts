export interface LanceriaJobsGetPayload {
  jobId: number
  amount: string
  employerAddress: string
  freelancerAddress: string
}
