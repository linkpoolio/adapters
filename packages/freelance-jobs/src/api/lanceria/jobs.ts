import { IJobs } from '../base'
import type { IJob } from '../../models/job'
import Job from '../../models/job'
import { Provider } from '../constants'

export default (fetch): IJobs => ({
  get: async ({ jobId }): Promise<IJob> => {
    const response = await fetch({
      url: `jobRequests/`,
      params: {
        jobId,
      },
    })
    return Job.Single(response.data, Provider.LANCERIA)
  },
})
