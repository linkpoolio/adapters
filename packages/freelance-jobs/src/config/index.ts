import { Requester } from '@chainlink/ea-bootstrap'
import { Config } from '@chainlink/types'

export const NAME = 'FREELANCE-JOBS'

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix)
  return config
}
