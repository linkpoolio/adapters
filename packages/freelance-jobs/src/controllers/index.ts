export * as jobs from './jobs'
