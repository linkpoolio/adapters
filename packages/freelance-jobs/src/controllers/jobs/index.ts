import execute from './controller'
import { inputParameters } from './input'

const supportedEndpoints = ['jobs']
const endpointResultPaths = {}
const description = 'This endpoint returns a single job.'

export { supportedEndpoints, endpointResultPaths, description, inputParameters, execute }
