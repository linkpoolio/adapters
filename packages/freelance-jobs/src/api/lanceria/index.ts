import { Requester } from '@chainlink/ea-bootstrap'
import { Config, RequestConfig } from '@chainlink/types'
import { injector } from '@linkpool/shared'

import { Base } from '../base'
import jobs from './jobs'
import { Provider as ProviderName, providerToBaseUrl } from '../constants'

const Fetch = (config: Config) => {
  config.api.baseURL = providerToBaseUrl.get(ProviderName.LANCERIA) as string
  config.api.headers['x-api-key'] = config.apiKey

  return async ({ params, url }: RequestConfig): Promise<any> => {
    const options = {
      ...config.api,
      url,
      params,
    }
    return Requester.request<any>(options)
  }
}

const Provider = (config: Config): Base => {
  const fetch = Fetch(config)
  const endpoints = {
    jobs,
  }

  return injector(endpoints, fetch)
}

export default Provider
