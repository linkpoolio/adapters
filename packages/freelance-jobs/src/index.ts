import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute, endpointSelector } from './adapter'
import * as endpoints from './controllers'
import { makeConfig, NAME } from './config'

const adapterContext = { name: NAME }

const { server } = expose(adapterContext, makeExecute(), undefined, endpointSelector)
export { NAME, makeExecute, makeConfig, server, endpoints }
