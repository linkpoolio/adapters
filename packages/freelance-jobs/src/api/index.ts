import { AdapterError, util } from '@chainlink/ea-bootstrap'
import { Config } from '@chainlink/types'

import { Provider } from './constants'
import lanceria from './lanceria'

export default (config: Config) => {
  const provider = util.getRequiredEnv('API_PROVIDER')

  switch (provider) {
    case Provider.LANCERIA:
      return lanceria(config)
    default: {
      const message = `Unsupported provider: ${provider}. Check API_PROVIDER env var`
      throw new AdapterError({
        statusCode: 200,
        message,
        cause: message,
      })
    }
  }
}
