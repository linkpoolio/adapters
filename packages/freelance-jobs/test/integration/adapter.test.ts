import http from 'http'
import process from 'process'
import type { AddressInfo } from 'net'
import request from 'supertest'
import type { SuperTest, Test } from 'supertest'

import { testJobs } from './lanceria/jobs'
import { server as startServer } from '../../src'

let oldEnv: NodeJS.ProcessEnv

export interface SuiteContext {
  server: http.Server
  req: SuperTest<Test>
}

beforeAll(() => {
  oldEnv = JSON.parse(JSON.stringify(process.env))
  process.env.API_VERBOSE = 'true'
  process.env.CACHE_ENABLED = 'false'
  process.env.WARMUP_ENABLED = 'false'
  process.env.API_PROVIDER = 'lanceria'
  process.env.API_KEY = 'fake-api-key'
})

afterAll(() => {
  process.env = oldEnv
})

describe('execute', () => {
  const context: SuiteContext = {
    server: null,
    req: null,
  }

  beforeEach(async () => {
    context.server = await startServer()
    context.req = request(`localhost:${(context.server.address() as AddressInfo).port}`)
  })

  afterEach((done) => {
    context.server.close(done)
  })

  describe('lanceria', () => {
    describe('jobs endpoint', () => testJobs(context))
  })
})
