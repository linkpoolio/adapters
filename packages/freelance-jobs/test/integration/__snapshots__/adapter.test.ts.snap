// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute lanceria jobs endpoint method: get returns a single job 1`] = `
Object {
  "data": Object {
    "employer": Object {
      "address": "0x6613A30A0D7F4011Cb569ca89f5a107cF9A542f2",
    },
    "freelancer": Object {
      "address": "0xbDf9CD30F6201B02F48d94878a86cf9B375f6344",
    },
    "id": 1,
    "payment": Object {
      "amount": "10000000000000000000",
      "currency": "LANC",
      "units": "wei",
    },
  },
  "jobRunID": 1,
  "statusCode": 200,
}
`;

exports[`execute lanceria jobs endpoint method: get returns an error response when the API errors too 1`] = `
Object {
  "error": Object {
    "feedID": "{\\"data\\":{\\"endpoint\\":\\"jobs\\",\\"method\\":\\"get\\",\\"jobId\\":1}}",
    "message": "Unexpected error requesting the 'jobs' endpoint. Reason: AdapterError: Request failed with status code 500",
    "name": "AdapterError",
    "url": "https:/api-testnet.lanceria.com/jobRequests/",
  },
  "jobRunID": 1,
  "providerStatusCode": 500,
  "status": "errored",
  "statusCode": 200,
}
`;
